ibis/backends/dask/client.py

```diff
@@ -0,0 +1,468 @@
+from dask.dataframe.core import DataFrame
+import dateutil.parser
+import numpy as np
+import pandas as pd
+import pytz
+import toolz
+from multipledispatch import Dispatcher
+from pandas.api.types import CategoricalDtype, DatetimeTZDtype
+from pkg_resources import parse_version
+
+import ibis.client as client
+import ibis.common.exceptions as com
+import ibis.expr.datatypes as dt
+import ibis.expr.operations as ops
+import ibis.expr.schema as sch
+import ibis.expr.types as ir
+
+from .core import execute_and_reset
+
+try:
+    infer_dask_dtype = pd.api.types.infer_dtype
+except AttributeError:
+    infer_dask_dtype = pd.lib.infer_dtype
+
+
+_ibis_dtypes = toolz.valmap(
+    np.dtype,
+    {
+        dt.Boolean: np.bool_,
+        dt.Null: np.object_,
+        dt.Array: np.object_,
+        dt.String: np.object_,
+        dt.Binary: np.object_,
+        dt.Date: 'datetime64[ns]',
+        dt.Time: 'timedelta64[ns]',
+        dt.Timestamp: 'datetime64[ns]',
+        dt.Int8: np.int8,
+        dt.Int16: np.int16,
+        dt.Int32: np.int32,
+        dt.Int64: np.int64,
+        dt.UInt8: np.uint8,
+        dt.UInt16: np.uint16,
+        dt.UInt32: np.uint32,
+        dt.UInt64: np.uint64,
+        dt.Float32: np.float32,
+        dt.Float64: np.float64,
+        dt.Decimal: np.object_,
+        dt.Struct: np.object_,
+    },
+)
+
+
+_numpy_dtypes = toolz.keymap(
+    np.dtype,
+    {
+        'bool': dt.boolean,
+        'int8': dt.int8,
+        'int16': dt.int16,
+        'int32': dt.int32,
+        'int64': dt.int64,
+        'uint8': dt.uint8,
+        'uint16': dt.uint16,
+        'uint32': dt.uint32,
+        'uint64': dt.uint64,
+        'float16': dt.float16,
+        'float32': dt.float32,
+        'float64': dt.float64,
+        'double': dt.double,
+        'unicode': dt.string,
+        'str': dt.string,
+        'datetime64': dt.timestamp,
+        'datetime64[ns]': dt.timestamp,
+        'timedelta64': dt.interval,
+        'timedelta64[ns]': dt.Interval('ns'),
+    },
+)
+
+
+_inferable_dask_dtypes = {
+    'boolean': dt.boolean,
+    'string': dt.string,
+    'unicode': dt.string,
+    'bytes': dt.string,
+    'empty': dt.string,
+}
+
+
+@dt.dtype.register(np.dtype)
+def from_numpy_dtype(value):
+    try:
+        return _numpy_dtypes[value]
+    except KeyError:
+        raise TypeError(
+            'numpy dtype {!r} is not supported in the dask backend'.format(
+                value
+            )
+        )
+
+
+@dt.dtype.register(DatetimeTZDtype)
+def from_dask_tzdtype(value):
+    return dt.Timestamp(timezone=str(value.tz))
+
+
+@dt.dtype.register(CategoricalDtype)
+def from_dask_categorical(value):
+    return dt.Category()
+
+
+@dt.infer.register(
+    (np.generic,)
+    + tuple(
+        frozenset(
+            np.signedinteger.__subclasses__()
+            + np.unsignedinteger.__subclasses__()  # np.int64, np.uint64, etc.
+        )
+    )  # we need this because in Python 2 int is a parent of np.integer
```

Comment:
can we simplify (e.g. no python 2) or is that just copied?

Reply:
This is just copied (from https://github.com/ibis-project/ibis/blob/master/ibis/backends/pandas/client.py#L120) - I can update it in both places.

Reply:
:->